### ISSUE TITLE
Optimize SparseSet::insert (see PERF comment on insert)

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#1557]
From Future Work of #1525.

**Comment by alice-i-cecile** [comment]
Comment in question:
```rust
        // PERF: switch to this. it's faster but it has an invalid memory access on
        // table_add_remove_many let dense = &mut self.dense;
        // let indices = &mut self.indices;
        // let dense_index = *self.sparse.get_or_insert_with(index.clone(), move || {
        //     if dense.len() == dense.capacity() {
        //         dense.reserve(64);
        //         indices.reserve(64);
        //     }
        //     let len = dense.len();
        //     // SAFE: we set the index immediately
        //     unsafe {
        //         dense.set_len(len + 1);
        //         indices.set_len(len + 1);
        //     }
        //     len
        // });
        // // SAFE: index either already existed or was just allocated
        // unsafe {
        //     *self.dense.get_unchecked_mut(dense_index) = value;
        //     *self.indices.get_unchecked_mut(dense_index) = index;
        // }
```

Found here: https://github.com/bevyengine/bevy/blob/172f4d6d787059f9d42fc262e9f766f26ac45e20/crates/bevy_ecs/src/storage/sparse_set.rs#L274

**Comment by alice-i-cecile** [comment]
> On the topic of issues, https://github.com/bevyengine/bevy/issues/1557 should probably be closed. This PR removes a required API and removes the commented out code. SparseSet::insert isn't a particularly hot API needing this, though I'm sure it was when ECS V2 was introduced, ComponentSparseSet has taken that role, and applying the same optimization might be possible, though harder to prove correct.

Closed per #5065.
